src/http.c

```diff
@@ -844,9 +885,29 @@ static int parse_xml_config(struct tunnel *tunnel, const char *buffer)
 		}
 	}
 
+	Node *route = tunnel->config->routes;
+	while(route){
```

Comment:
In the code creating the nodes, there is

    newNode->next = cfg->routes;
    cfg->routes = newNode;

Using `while(route)` assumes there will be a null pointer somewhere. But for me it seems the last item loops back to the beginning.